### PR TITLE
Bug 1839101: Namespaced navigation item to non-resource pages

### PR DESCRIPTION
**Pull request by fmartingr**
## Summary
Currently the navigation from the console works well when the target comes from a resource in the cluster, but some views that depends on namespaces that are not resources (like topology) use direct links that need to handle the namespace the user is on internally.

This causes a mild bug on the console when you have two tabs open on different namespaces, since these links use the local storage to determine the namespace the user is on and that would be the last one the user accessed.

## Proposal

Created a new navigation item that operates using a name and a namespace in the form of: `<view>/ns/<namespace>` which is the format of the navigation items replaced with this.

## Pending
- [ ] Naming for the component (Used `ViewNS`).
- [ ] Paths for these kind of pages (Used `/<name>/ns/<namespace>`) but not sure all pages use this.
- [ ] Path generation requires a namespace and not all pages may support `all-namespaces`.

## Notes

- I tried to follow the guidelines as shown in the project and the docs, if I missed something please let me know.
- I have full-stack experience on several languages, not a lot with React though, but eager to learn.

### CODE REVIEW

**Comment by christianvogt** [comment]
Thanks @fmartingr for your contribution.

@andrewballantyne @rohitkrai03 could you please have a look at this?
What needs to be thought of is whether a standard format of `/${name}/ns/${namespace}` is sufficient or whether we need more flexibility for extensions.

**Comment by fmartingr** [comment]
Hey @andrewballantyne, just pushed some changes. There were a bit more occurrences of the `HrefLink` on several files, I hope I understood everything well. Tests are passing and I went through the console myself to do some manual checks, but I'm sure there are some links that I don't see on the local console (like the metal3 one).

I've split the plugin changes in different commits in case I need to revert something.

Having everything linted properly took a while, my eslint plugin in VSCode is not working properly, I need to take a look at that.

One question, should I commit the `yarn.lock` changes? The test script urges me to but I'm not sure I have to or this is done only for releases, to avoid conflicts.

**Comment by andrewballantyne** [comment]
@fmartingr Thanks for the update. I'll try to do a review today if I can find a few moments. The e2e tests look like they failed for a possibly unrelated change, but I will need to look into your latest changes to find out more.

**Comment by andrewballantyne** [comment]
> One question, should I commit the `yarn.lock` changes? The test script urges me to but I'm not sure I have to or this is done only for releases, to avoid conflicts.

Yup, `yarn.lock` (or the more native one, that we don't use, `package-lock.json`) need to be committed. If you changed the versions on any dependencies, the lock files are there to avoid any minor patch movements in the very large dependency tree. It's mostly there to prevent the "it works on my machine" problems around 3rd party libraries.

**Comment by fmartingr** [comment]
@andrewballantyne It seems that I had a `yarn.lock` where it shouldn't be (from a test command ran in an incorrect path it seems). I deleted it and the test command no longer shows up the message. No rush here, I will keep an eye on the issue for updates, thanks for your time!

**Comment by andrewballantyne** [comment]
/hold

Not going into 4.6, should go into 4.7 asap though.

File: frontend/packages/dev-console/src/plugin.tsx [hunk 1]
```diff
@@ -164,6 +166,22 @@ const plugin: Plugin<ConsumedExtensions> = [
...
     },
   },
+  {
+    type: 'NavItem/Href',
+    properties: {
+      perspective: 'dev',
+      group: 'top',
+      componentProps: {
+        name: 'GitOps',
+        namespaced: true,
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/plugin.tsx:176]: Gitops page is actually not a namespaced page.
```suggestion
        namespaced: false,
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Comment by jerolimov** [comment]
Hi @fmartingr, I'm apologize for the new delay. We are still interested in your PR. Do you see the option to rebase your changes on the latest master? Sorry for this extra round.

**Comment by christianvogt** [comment]
@jerolimov let's take over this PR and make sure we complete the work. We failed to followup on this PR.

**Comment by fmartingr** [comment]
Hey folks, I was on PTO last week, do you still need me to rebase this or are you taking care of it now?

**Comment by jerolimov** [comment]
Thanks @fmartingr, we will pick your changes. Thanks a lot for your work!

**Comment by fmartingr** [comment]
Happy to be of assistance @jerolimov

**Comment by jerolimov** [comment]
I created a new PR with the change of this PR and marked you as author of this change. I still need to check the solution myself now. Big thanks @fmartingr again!

/close

**Comment by jerolimov** [comment]
Hey @fmartingr, finally we got #8249 merged and [Bugzilla bug 1839101](https://bugzilla.redhat.com/show_bug.cgi?id=1839101) closed! :partying_face::partying_face::partying_face: 

Sorry again for the long delay and big thanks for your contribution!!

**Comment by fmartingr** [comment]
Glad to know! Thanks @jerolimov :)

**Pull request closed without merging**